core/domain/exp_jobs_one_off.py

```diff
@@ -413,12 +412,30 @@ def map(item):
                         'revert_version_number'):
                     reverted_version = change_list[0]['version_number']
                     # pylint: disable=line-too-long
-                    exp_services.create_and_save_state_id_mapping_model_for_reverted_exploration(
-                        exploration.id, exploration.version - 1, reverted_version)
+                    state_id_mapping = exp_services.generate_state_id_mapping_model_for_reverted_exploration(
+                        exploration.id, exploration.version - 1,
+                        reverted_version)
                     # pylint: enable=line-too-long
                 else:
-                    exp_services.create_and_save_state_id_mapping_model(
-                        exploration, change_list)
+                    state_id_mapping = (
+                        exp_services.generate_state_id_mapping_model(
+                            exploration, change_list))
+
+                # pylint: disable=protected-access
+                instance_id = (
```

Comment:
Use of private methods is a red flag. Why not just use exp_models.StateIdMappingModel.create()?

Reply:
Done. Added new `overwrite` argument to allow/disallow overwrites.